plugins/GSdx/GSTextureCacheSW.cpp

```diff
@@ -133,30 +121,26 @@ void GSTextureCacheSW::RemoveAll()
 
 void GSTextureCacheSW::IncAge()
 {
-	for(hash_set<Texture*>::iterator i = m_textures.begin(); i != m_textures.end(); )
+	for(auto i = m_textures.begin(); i != m_textures.end(); )
 	{
-		hash_set<Texture*>::iterator j = i++;
-
-		Texture* t = *j;
+		Texture* t = *i;
 
 		if(++t->m_age > 10)
 		{
-			m_textures.erase(j);
+			i = m_textures.erase(i);
```

Comment:
Sorry, I read the wrong line on the man. Anyway, I still think previous pattern is better. It avoid the else branch and it is more generic. Tbh, a better solution could be something with algortihm/lambda (speed is useless here)